client/post-editor/editor-categories-tags/accordion.jsx

```diff
@@ -1,88 +1,74 @@
-
-module.exports = React.createClass( {
-	displayName: 'EditorCategoriesTagsAccordion',
-
-	propTypes: {
-		site: React.PropTypes.object,
-		post: React.PropTypes.object,
-		// passed down from CategoryListData
-		categories: React.PropTypes.array,
-		categoriesFound: React.PropTypes.number,
-		categoriesHasNextPage: React.PropTypes.bool,
-		categoriesFetchingNextPage: React.PropTypes.bool,
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPostValue } from 'state/posts/selectors';
+import { getSiteOption } from 'state/sites/selectors';
+import { getTerm } from 'state/terms/selectors';
+
+export class EditorCategoriesTagsAccordion extends Component {
+	static propTypes = {
+		site: PropTypes.object,
+		post: PropTypes.object,
+		translate: PropTypes.func,
+		postTerms: PropTypes.object,
+		postType: PropTypes.string,
+		defaultCategory: PropTypes.object,
+		defaultCategoryId: PropTypes.number,
 		// passed down from TagListData
-		tags: React.PropTypes.array,
-		tagsHasNextPage: React.PropTypes.bool,
-		tagsFetchingNextPage: React.PropTypes.bool
-	},
-
-	isCategoriesLoading: function() {
-		if ( ! this.props.site ) {
-			return true;
-		}
-
-		return this.props.categoriesFetchingNextPage;
-	},
-
-	getCategories: function() {
-		var defaultCategory;
-
-		if ( ! this.props.post ) {
-			return [];
-		}
-
-		if ( this.props.post.category_ids ) {
-			return this.props.post.category_ids;
-		}
-
-		defaultCategory = siteUtils.getDefaultCategory( this.props.site );
-		if ( defaultCategory ) {
-			return [ defaultCategory ];
-		}
-
-		return [];
-	},
-
-	renderCategories: function() {
-		if ( isPage( this.props.post ) ) {
+		tags: PropTypes.array,
+		tagsHasNextPage: PropTypes.bool,
+		tagsFetchingNextPage: PropTypes.bool
+	};
+
+	renderCategories() {
+		const { translate, postType, defaultCategory, defaultCategoryId, siteId } = this.props;
+		if ( postType === 'page' ) {
 			return;
 		}
+		// If the default category is not in the state tree, and is not Uncategorized, fetch via QueryTerms
+		const fetchDefaultCategory = defaultCategoryId && defaultCategoryId !== 1 && ( defaultCategory === null );
 
 		return (
 			<AccordionSection>
-				<label className="editor-drawer__label">
-					<span className="editor-drawer__label-text">
-						{ this.translate( 'Categories' ) }
-						<InfoPopover position="top left">
-							{ this.translate( 'Use categories to group your posts by topic.' ) }
-						</InfoPopover>
-					</span>
-				</label>
-				<Categories site={ this.props.site } post={ this.props.post } />
+				<EditorDrawerLabel>
+					{ translate( 'Categories' ) }
+					<InfoPopover position="top left">
+						{ translate( 'Use categories to group your posts by topic.' ) }
+					</InfoPopover>
+				</EditorDrawerLabel>
+				{ fetchDefaultCategory &&
+					<QueryTerms
+						siteId={ siteId }
+						taxonomy="category"
+						query={ { ID: defaultCategoryId } }
```

Comment:
`ID` is not a valid query parameter for the terms endpoint:

https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/taxonomies/%24taxonomy/terms/

Maybe this was working for you because the default category was returned in the first page of results?